### ISSUE TITLE
V4L2 decoder does not correctly implement VIDIOC_G_FMT and VIDIOC_S_FMT

### ISSUE THREAD

**Issue by EwanRoycroft** [01org/libyami#888]
According to the [V4L2 documentation](https://www.linuxtv.org/downloads/v4l-dvb-apis-new/userspace-api/v4l/vidioc-g-fmt.html), `VIDIOC_G_FMT` is exclusive to capture devices, and you can use this method with other drivers to get information about the output format; yet yami [limits it](https://github.com/intel/libyami/blob/edd1ad15e383d32c8d67bdc9198c834b6acebca5/v4l2/v4l2_decode.cpp#L1159) to `V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE`.

The documentation also says of `VIDIOC_S_FMT` that:
> "Drivers should not return an error code unless the type field is invalid,"

and that, after setting the format as requested:
> "the VIDIOC_S_FMT ioctl returns the current format parameters as VIDIOC_G_FMT does."

I have found that this is not the case with yami. If certain fields are left empty, it will [throw an error](https://github.com/intel/libyami/blob/edd1ad15e383d32c8d67bdc9198c834b6acebca5/v4l2/v4l2_decode.cpp#L1113) instead of filling them with the current values.

**Comment by xuguangxin** [comment]
yes. we are not fully aligned with spec. the spec original designed for capture or output device. some items may not reasonable to decode or encoder. 
Are you expect this VIDIOC_S_FMT/VIDIOC_G_FMT  behavior in our application?
thanks

**Comment by EwanRoycroft** [comment]
I have a program that uses the V4L2 driver on the Raspberry Pi to decode video. Assuming other drivers follow the spec, it should work cross-platform; but it fails on yami when making calls to `VIDIOC_G_FMT` / `VIDIOC_S_FMT`.
